make test fails one test

ok      github.com/dipperin/dipperin-core/core/csbft/state-machine      (cached)        coverage: 92.0% of statements
t=2019-04-15T21:20:36+0800 lvl=info msg="start env: "
--- FAIL: TestNodeConfig_IpcEndpoint (0.00s)
    config_test.go:107: 
                Error Trace:    config_test.go:107
                Error:          Not equal: 
                                expected: "/tmp/path"
                                actual  : "/var/folders/7z/46b7wbbn7dl1by7bhvpv9j_80000gn/T/path"
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -/tmp/path
                                +/var/folders/7z/46b7wbbn7dl1by7bhvpv9j_80000gn/T/path
                Test:           TestNodeConfig_IpcEndpoint
t=2019-04-15T21:20:36+0800 lvl=info msg="get stored genesis hash" hash=0x0000000000000000000000000000000000000000000000000000000000000000
t=2019-04-15T21:20:36+0800 lvl=info msg="set genesis registerDB successful" root=0x19da3f2bf161a3a674df2eeec596b4845806e0b411745a99b29fda56554bec8d
t=2019-04-15T21:20:36+0800 lvl=info msg="write genesis block" hash=0x2d3a91e4d531f11dd1d440a73fa4451135b5e7cef022eedfc2c09fb09ec8fb2a
t=2019-04-15T21:20:36+0800 lvl=dbug msg="Persisted trie from memory database" nodes=13 size="1.32 kB" time=67.276µs gcnodes=0 gcsize="0.00 B" gctime=0s livenodes=1 livesize="0.00 B"
t=2019-04-15T21:20:36+0800 lvl=dbug msg="Persisted trie from memory database" nodes=371 size="36.99 kB" time=1.082567ms gcnodes=0 gcsize="0.00 B" gctime=0s livenodes=1 livesize="0.00 B"
t=2019-04-15T21:20:36+0800 lvl=dbug msg="Persisted trie from memory database" nodes=3 size="251.00 B" time=16.246µs gcnodes=0 gcsize="0.00 B" gctime=0s livenodes=1 livesize="0.00 B"
t=2019-04-15T21:20:36+0800 lvl=info msg="genesis early contract check success"
t=2019-04-15T21:20:36+0800 lvl=info msg="GetPBFTPrimaryNode# The current height on the chain is the last block of the round, and verifiers of the next round should be taken."
["enode://b832f4f2fe19dbc5604766bbb268a6d0f7ce9ce381b034b262a92f0ad8283a1b5fa058dea5269b66fbb2014a24fa7198c6dc2d8c9cbac7a348258fc20702561f@127.0.0.1:10003","","node"]
t=2019-04-15T21:20:36+0800 lvl=eror msg="Node URL node: invalid URL scheme, want \"enode\"\n"
t=2019-04-15T21:20:36+0800 lvl=eror msg="Can't load node file /Users/qingfengcai: read /Users/qingfengcai: is a directory"
t=2019-04-15T21:20:36+0800 lvl=info msg="can't load nodekey from data dir, gen a new key" key file path=nodekey load err="open nodekey: no such file or directory"
t=2019-04-15T21:20:36+0800 lvl=info msg="call node stop"
t=2019-04-15T21:20:36+0800 lvl=info msg="the err service is: " service={err:0xc000168cb0}
t=2019-04-15T21:20:36+0800 lvl=eror msg="start node service failed" err="test error"
t=2019-04-15T21:20:36+0800 lvl=info msg="call node stop"
FAIL
coverage: 38.5% of statements
FAIL    github.com/dipperin/dipperin-core/core/dipperin 0.405s

--- a/core/dipperin/config_test.go
+++ b/core/dipperin/config_test.go
@@ -102,6 +102,7 @@ func TestNodeConfig_IpcEndpoint(t *testing.T) {
 	nodeConfig = NodeConfig{
 		IPCPath: "path",
 	}
+
 	result = nodeConfig.IpcEndpoint()
 	if runtime.GOOS == "windows" {
 		assert.Equal(t, `\\.\pipe\`+nodeConfig.IPCPath, result)

--- a/core/dipperin/config_test.go
+++ b/core/dipperin/config_test.go
@@ -20,6 +20,8 @@ import (
 	"github.com/stretchr/testify/assert"
 	"runtime"
 	"testing"
+	"path/filepath"
+	"os"
 )
 
 func TestNodeConfig_FullChainDBDir(t *testing.T) {
@@ -104,7 +106,7 @@ func TestNodeConfig_IpcEndpoint(t *testing.T) {
 	if runtime.GOOS == "windows" {
 		assert.Equal(t, `\\.\pipe\`+nodeConfig.IPCPath, result)
 	} else {
-		assert.Equal(t, "/tmp/path", result)
+		assert.Equal(t, filepath.Join(os.TempDir(), nodeConfig.IPCPath), result)
 	}
 
 	nodeConfig = NodeConfig{

